kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/SolverServiceBase.java

```diff
@@ -515,4 +516,41 @@ private void terminateSolverEarly(SolverInstanceContext sic) {
         }
         sic.getSolver().terminateEarly();
     }
+    
+    protected boolean boostrapContinuousPlanning(String containerId, List<ContinuousPlanning> plannings) {
+        for (ContinuousPlanning planning : plannings) {
+            if (planning.accept(containerId)) {                
+                logger.info("ContinuousPlanning {} is going to be used for container {}", planning, containerId);
+                
+                String solverId = planning.getSolverId();
+                SolverInstance instance = new SolverInstance();
+                instance.setSolverConfigFile(solverId + "-solver.xml");
+                instance.setContainerId(containerId);
+                instance.setSolverId(solverId);
+                
+                logger.debug("About to start solver {}", instance);
+                createSolver(containerId, solverId, instance);
+                
+                logger.debug("Solver with id {} created successfully", solverId);
+                SolverInstanceContext solverInstance = solvers.get(solverId);
+                
+                Solver<Object> solver = solverInstance.getSolver();
+                SolverEventListener<Object> eventListener = planning.getCallback();
+                solver.addEventListener(eventListener);
+                logger.debug("Added solver event listner {} to solver {}", eventListener, solver);
+                
+                List<ProblemFactChange<Object>> problemFacts = planning.loadFacts();
+                logger.debug("About to add following problem fact changes {} to solver {}", problemFacts, solverId);
+                if (!problemFacts.isEmpty()) {
+                    solver.addProblemFactChanges(problemFacts);
+                }
+                
+                solver.solve(planning.getPlanningProblem());
```

Comment:
Let's use SolverServiceBase.executor for boostrapContinuousPlanning() to avoid client timeouts. See SolverServiceBase.solvePlanningProblem for a reference.